pkg/instancetype/expand/expand.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+package expand
+
+import (
+	"fmt"
+
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/defaults"
+	"kubevirt.io/kubevirt/pkg/instancetype/apply"
+	instancetypeErrors "kubevirt.io/kubevirt/pkg/instancetype/errors"
+	"kubevirt.io/kubevirt/pkg/instancetype/find"
+	preferenceFind "kubevirt.io/kubevirt/pkg/instancetype/preference/find"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+	utils "kubevirt.io/kubevirt/pkg/util"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+type Expander struct {
+	clusterConfig      *virtconfig.ClusterConfig
+	vmiApplier         *apply.VMIApplier
+	instancetypeFinder *find.SpecFinder
+	preferenceFinder   *preferenceFind.SpecFinder
+}
+
+func New(
```

Comment:
For consistency with e.g. appliers it should be `NewExpander`. WDYT?

Reply:
I've used `NewFooer` names where there are conflicts with additional `New` functions within a package, for example in the find packages where multiple implementations are provided. In this case there's only a single implementation so I think `New` is okay here.